packages/icons/generateIcons.js

```diff
@@ -0,0 +1,121 @@
+const fs = require('fs');
+const path = require('path');
+const prettier = require('prettier');
+const { optimize } = require('svgo');
+
+const iconAssetsDirname = 'assets/icons';
+const flagAssetsDirname = 'assets/flags';
+const cryptoIconAssetsDirname = 'assets/cryptoIcons';
+const iconsFilePath = './src/icons.ts';
+
+// https://github.com/svg/svgo#built-in-plugins
+const svgoConfig = {
+    multipass: true,
+    js2svg: {
+        indent: 2, // string with spaces or number of spaces. 4 by default
+        pretty: true, // boolean, false by default
+    },
+    plugins: [
+        {
+            name: 'preset-default',
+            params: {
+                overrides: {
+                    removeViewBox: false,
+                },
+            },
+        },
+        'removeXMLNS',
+        'prefixIds',
+        // it's necessary to remove all dimension tags to allow resizing
+        'removeDimensions',
+        'removeRasterImages',
+        'removeScriptElement',
+    ],
+};
+
+const optimizeSvgAssets = assetsDirname => {
+    const assetsDir = path.join(__dirname, assetsDirname);
+    const assetFileNames = fs.readdirSync(assetsDir);
+    return assetFileNames
+        .map(fileName => ({
+            fileName,
+            content: fs.readFileSync(path.resolve(assetsDir, fileName)).toString(),
+        }))
+        .map(({ fileName, content }) => ({
+            fileName,
+            content: optimize(content, svgoConfig).data,
+        }));
+};
+
+const generateIconsFileContent = assetTypesArray => {
+    const mappedAssetTypes = assetTypesArray.map(
+        ({ name, assets, dirname, typeName }) => `
+           export const ${name} = {
+            ${assets
+                .map(
+                    ({ fileName }) =>
+                        `${fileName.replace('.svg', '')}: require('../${dirname}/${fileName}')`,
+                )
+                .join(',')}
+        } as const;
+        export type ${typeName} = keyof typeof ${name};
+       `,
+    );
+    return `
+    /* eslint-disable global-require */
+
+    // !!! IMPORTANT: This file is autogenerated !!!
+    // If you want to add of modify icons please read README.md to find out how to do it
+
+    ${mappedAssetTypes.join('')}
+    `;
+};
+
+const writeOptimizedAssets = assetTypesArray => {
+    assetTypesArray.forEach(({ assets, dirname }) => {
+        assets.forEach(({ fileName, content }) =>
+            fs.writeFileSync(path.resolve(dirname, fileName), content),
+        );
+    });
+};
+
+(async () => {
+    const icons = optimizeSvgAssets(iconAssetsDirname);
+    const cryptoIcons = optimizeSvgAssets(cryptoIconAssetsDirname);
+    const flagIcons = optimizeSvgAssets(flagAssetsDirname);
+
+    const assetTypes = [
```

Comment:
could be moved to top of file and you can probably kill dirname const and place it directly here